quinn-proto/src/crypto/types.rs

```diff
@@ -52,6 +52,14 @@ impl CertificateChain {
     pub fn from_certs(certs: impl IntoIterator<Item = Certificate>) -> Self {
         certs.into_iter().collect()
     }
+
+    /// Iterate over the certificates
+    pub fn iter<'a>(&'a self) -> impl Iterator<Item = Certificate> + 'a {
+        self.certs
+            .iter()
+            .cloned()
+            .map(|cert| Certificate { inner: cert })
+    }
```

Comment:
I'd prefer for this to either be turned into `into_iter` (consuming `self`) or to finagle things until it can become an `impl Iterator<Item = &'a Certificate>`.

Reply:
Why is that? Implementing `Iterator` does not seem to make sense here.

Reply:
The trait associated with `into_iter` is `IntoIterator`.

Reply:
Yeah, wasn't sure about what you meant by "finagle things until it can become an `impl Iterator`. So you mean `impl IntoIterator for &CertificateChain`? That sounds like a plan.

Reply:
I'm not sure I understand how I could turn an `impl Iterator<Item = &'a rustls::Certificate>` into an `impl Iterator<Item = &'a Certificate>`

Reply:
I meant `impl IntoIterator for CertificateChain`, or `fn iter(&self) -> impl Iterator<Item = &Certificate>`, as two reasonable alternatives.

Reply:
The latter option would involve storing a `Vec<Certificate>` or similar internally and converting for rustls's convenience as necessary. But I think the former is a good choice if it serves your needs.

Reply:
Ok thanks.

I was hesitant to make swiping changes, as `CertificateChain::certs` is `pub(crate)`. If that’s appreciated, I’ll try to be a bit more audacious next time :)